### PR TITLE
[#175971735] Add notice email redis cache

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/services/redisSessionStorage.ts [hunk 1]
```diff
@@ -47,6 +47,7 @@ const myPortalTokenPrefix = "MYPORTAL-";
 const bpdTokenPrefix = "BPD-";
 const userSessionsSetKeyPrefix = "USERSESSIONS-";
 const sessionInfoKeyPrefix = "SESSIONINFO-";
+const NoticeEmailPrefix = "NoticeEmail-";
```
Comment (cloudify) [inline src/services/redisSessionStorage.ts:50]:
```suggestion
const NoticeEmailPrefix = "NOTICEEMAIL-";
```

File: src/controllers/pagoPAController.ts [hunk 2]
```diff
@@ -38,27 +49,70 @@ export default class PagoPAController {
...
+                  if (response.kind !== "IResponseSuccessJson") {
+                    // if getProfile returns a failure, we just return it
+                    return fromLeft(response);
+                  }
+                  // if no validated email is provided into InitializedProfile
+                  // spid_email will be used for notice email
+                  const maybeNoticeEmail: EmailAddress | undefined =
+                    response.value.email && response.value.is_email_validated
+                      ? response.value.email
+                      : user.spid_email;
```
Comment (balanza) [inline src/controllers/pagoPAController.ts:81]: why undefined? can't we use an Option?

File: src/services/redisSessionStorage.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: Fixed in 5796035

**Review by AleDore — COMMENTED** [review]
lgtm, apart from minor change requested also by @balanza

File: src/controllers/pagoPAController.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/controllers/pagoPAController.ts:82]: Fixed in 36e2b68

**Pull request merged**